lib/ansible/modules/cloud/docker/docker_swarm_service.py

```diff
@@ -2689,8 +2688,13 @@ def main():
             ) is not None,
             usage_msg='set rollback_config.order'
         ),
+        resolve_image_is_true=dict(
+            docker_api_version='1.30',
+            docker_py_version='3.2.0',
+            detect_usage=lambda c: c.module.params['resolve_image'] is True,
```

Comment:
You don't need `detect_usage` and `usage_msg` here.

Reply:
So the version check will not be triggered if `resolve_image` is set to `false` as default? Though it was only skipped if the value is `None`.

Reply:
I guess an option would be to remove the default and let it be `None`.

Reply:
The version check will not be triggered if an option equals its default value, which is often `None`, but can also be something else. If it wouldn't be like this, a lot of minimal version check entries would need to be way more explicit. The relevant code is here:
https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/docker/common.py#L458-L460
So keeping `no` as a default is totally fine.

Reply:
Makes sense. Missed that line.